enterprise/reporting/client/client.go

```diff
@@ -92,10 +92,7 @@ func (c *Client) Send(ctx context.Context, payload any) error {
 		return err
 	}
 
-	u, err := url.JoinPath(c.reportingServiceURL, string(c.path))
-	if err != nil {
-		return err
-	}
+	u := strings.TrimSuffix(c.reportingServiceURL, "/") + string(c.path)
```

Comment:
can we use `url.JoinPath`?

Reply:
I wasn't happy with regressing to previous implementation either. Your comment nudge me, to come up with a cleaner approach.

I've move the URL construction [here](https://github.com/rudderlabs/rudder-server/pull/5842/files#diff-2c09416a82ca9dbb68294523ea370de02bfcfaab73c98501cc778d20717fe441R45).

* Applying URL parse for both base URL and service route. Properly handling query parameters.
* Rename `Path` to `ServiceRoute`, to semantically encapsulate query params.